src/main/java/seedu/address/logic/commands/customer/CustomerEditCommand.java

```diff
@@ -125,6 +137,51 @@ private static Customer createEditedCustomer(Customer customerToEdit,
                 updatedEmail, updatedAddress);
     }
 
+    /**
+     * Updates all the deliveries associated with the customer with new customer details.
+     */
+    protected static void updateDelivery(Model model, Customer editedCustomer) {
+        int customerId = editedCustomer.getCustomerId();
+        Stream<Delivery> deliveries = model.getDeliveryByCustomerId(customerId);
+        deliveries.forEach(d -> {
+            DeliveryEditDescriptor descriptor = new DeliveryEditDescriptor();
+            descriptor.setCustomerId(customerId);
+            Delivery editedDelivery = createEditedDelivery(d, descriptor, editedCustomer);
+            model.setDelivery(d, editedDelivery);
+        });
+    }
+
+    /**
+     * Creates and returns a {@code Delivery} with the customer details edited.
+     *
+     * @param deliveryToEdit         {@code Delivery} which the command edits.
+     * @param deliveryEditDescriptor {@code editDeliveryDescriptor} details to edit the delivery with.
+     * @param editedCustomer         {@code Customer} which the delivery is associated with.
+     */
+    private static Delivery createEditedDelivery(Delivery deliveryToEdit, DeliveryEditDescriptor
+            deliveryEditDescriptor, Customer editedCustomer) {
+
+        assert deliveryToEdit != null;
+
+        DeliveryName updatedDeliveryName =
+                deliveryEditDescriptor.getDeliveryName().orElse(deliveryToEdit.getName());
+
+        OrderDate orderDate = deliveryToEdit.getOrderDate();
+
+        DeliveryDate updatedDeliveryDate =
+                deliveryEditDescriptor.getDeliveryDate().orElse(deliveryToEdit.getDeliveryDate());
+
+        DeliveryStatus updatedDeliveryStatus =
+                deliveryEditDescriptor.getStatus().orElse(deliveryToEdit.getStatus());
+
+        Note updatedNote = deliveryEditDescriptor.getNote().orElse(deliveryToEdit.getNote());
```

Comment:
Actually would `deliveryEditDescriptor` be empty in this case since `descriptor` is a brand new `DeliveryEditDescriptor` so we can just directly set the variable? etc. 
`DeliveryName updatedDeliveryName = deliveryToEdit.getName()`

Reply:
Good catch, have updated it accordingly